src/main/java/gregtech/common/covers/detector/CoverDetectorFluidAdvanced.java

```diff
@@ -65,62 +70,60 @@ public void renderCover(@NotNull CCRenderState renderState, @NotNull Matrix4 tra
-        group.addWidget(
-                new CycleButtonWidget(10, 3 * (SIZE + PADDING), 4 * SIZE, SIZE, this::isInverted, this::setInverted,
-                        "cover.advanced_energy_detector.normal", "cover.advanced_energy_detector.inverted")
-                                .setTooltipHoverString("cover.generic.advanced_detector.invert_tooltip"));
-        group.addWidget(
-                new CycleButtonWidget(94, 3 * (SIZE + PADDING), 4 * SIZE, SIZE, this::isLatched, this::setLatched,
-                        "cover.generic.advanced_detector.continuous", "cover.generic.advanced_detector.latched")
-                                .setTooltipHoverString("cover.generic.advanced_detector.latch_tooltip"));
-
-        this.fluidFilter.initUI(5 + 4 * (SIZE + PADDING), group::addWidget);
-
-        return ModularUI.builder(GuiTextures.BACKGROUND, 176, 164 + 4 * (SIZE + PADDING))
-                .widget(group)
-                .bindPlayerInventory(player.inventory, GuiTextures.SLOT, 7, 164)
-                .build(this, player);
+    public ModularPanel buildUI(SidedPosGuiData guiData, PanelSyncManager guiSyncManager, UISettings settings) {
+        return GTGuis.defaultPanel(this)
+                .height(202)
+                .child(CoverWithUI.createTitleRow(getPickItem()))
+                .child(Flow.column()
+                        .top(28)
+                        .left(5).right(5)
+                        .child(createMinMaxRow("cover.advanced_fluid_detector.min",
+                                this::getMinValue, this::setMinValue,
+                                () -> " L", w -> w.setMaxLength(10)))
+                        .child(createMinMaxRow("cover.advanced_fluid_detector.max",
+                                this::getMaxValue, this::setMaxValue,
+                                () -> " L", w -> w.setMaxLength(10)))
+                        .child(Flow.row()
+                                .widthRel(1f)
+                                .coverChildrenHeight()
+                                .marginBottom(5)
+                                .child(new ToggleButton()
+                                        .size(72, 18)
+                                        .value(new BooleanSyncValue(this::isInverted, this::setInverted))
+                                        .addTooltipLine(IKey.lang("cover.generic.advanced_detector.invert_tooltip"))
+                                        .overlay(new DynamicDrawable(() -> {
+                                            String lang = "cover.advanced_energy_detector.";
+                                            lang += isInverted() ? "inverted" : "normal";
+                                            return IKey.lang(lang).style(TextFormatting.WHITE);
+                                        })))
+                                .child(new ToggleButton()
+                                        .size(72, 18)
+                                        .right(0)
```

Comment:
right 0? Does this even do anything?

Reply:
this puts the button to the right of the row